components/director/internal/domain/formation/service.go

```diff
@@ -459,6 +469,11 @@ func (s *service) AssignFormation(ctx context.Context, tnt, objectID string, obj
 	formationFromDB := ft.formation
 	switch objectType {
 	case graphql.FormationObjectTypeApplication, graphql.FormationObjectTypeRuntime, graphql.FormationObjectTypeRuntimeContext:
+		// If we assign it to the label definitions when it is in deleting state we risk leaving incorrect data
+		// in the LabelDefinition and formation assignments and failing to delete the formation later on
+		if formationFromDB.State == model.DeletingFormationState {
+			return nil, fmt.Errorf("cannot assign to formation with ID %q as it is in %q state", formationFromDB.ID, formationFromDB.State)
```

Comment:
I am not sure if that is a problem with INITIAL, READY or CREATE_ERROR. Only when in DELETE_ERROR or DELETING it could pose to be a problem as  the deletion would be attempted, but have failed, after that if it is retried and something is added, it would fail. Maybe for `DELETE_ERROR` assign should also not be allowed.

Reply:
As discussed offline, the DELETE_ERROR should be added as well. About the CREATE_ERROR both scenarios are valid - to not allow assignments if the formation is in CREATE_ERROR as well as to allow assignments in such cases. We need to follow up on what scenarios we have(or we'll have in the future) to determine if we need this check.

Reply:
As discussed offline, the DELETE_ERROR should be added as well. About the CREATE_ERROR both scenarios are valid - to not allow assignments if the formation is in CREATE_ERROR as well as to allow assignments in such cases. We need to follow up on what scenarios we have(or we'll have in the future) to determine if we need this check.